raiden/tests/unit/transfer/mediated_transfer/test_mediation_fee.py

```diff
@@ -71,24 +77,28 @@ def test_linspace():
 
 
 def test_rebalancing_fee_calculation():
-    max_imbalance_fee = FA(10 ** 18)
-    sample = calculate_imbalance_fees(TA(200), max_imbalance_fee)
+    sample = calculate_imbalance_fees(TA(200), RFA(500_000))  # 50%
     assert sample is not None
     assert len(sample) == NUM_DISCRETISATION_POINTS
+    assert all(0 <= x <= 200 for x, _ in sample)
     assert max(x for x, _ in sample) == 200
-    assert max(y for _, y in sample) == 10 ** 18
+    assert all(0 <= y <= 50 for _, y in sample)
+    assert max(y for _, y in sample) == 50  # 50% of the 100 TA per channel side
 
-    sample = calculate_imbalance_fees(TA(10), max_imbalance_fee)
+    sample = calculate_imbalance_fees(TA(10), RFA(200_000))  # 20%
     assert sample is not None
     assert len(sample) == 11
+    assert all(0 <= x <= 10 for x, _ in sample)
     assert max(x for x, _ in sample) == 10
-    assert max(y for _, y in sample) == 10 ** 18
+    assert all(0 <= y <= 1 for _, y in sample)
+    assert max(y for _, y in sample) == 1  # 20% of the 5 TA per channel side
 
-    sample = calculate_imbalance_fees(TA(1), max_imbalance_fee)
+    sample = calculate_imbalance_fees(TA(1), RFA(1_000_000))  # 100%
     assert sample is not None
     assert len(sample) == 2
+    assert all(0 <= x <= 1 for x, _ in sample)
     assert max(x for x, _ in sample) == 1
-    assert max(y for _, y in sample) == 10 ** 18
+    assert all(y == 0 for _, y in sample)
```

Comment:
It's hard for me to expect all zero here after seeing 100%.  Maybe I'll figure out why after reading more.

Reply:
So this is a rounding issue. I'm not sure what to do. I guess in general the case with 1 token wei is not common, but I wanted it to be stated explicitly in the test.

- `floor` and `int` and `round` give the same result
- `ceil` would return 1.

I guess it's best to return the division by two above, to make this better in this extreme case.

Reply:
Oh I was not seeing it's capacity==1 token wei.